### PR TITLE
Introducing an AWS2 support extension

### LINKED ISSUE #1092: Create a support extension for AWS2

### PR DESCRIPTION
**Pull request by oscerd**
Fixes #1092

### CODE REVIEW

File: extensions/aws2-s3/deployment/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/Aws2S3Processor.java [hunk 1]
```diff
@@ -89,12 +89,6 @@ void client(BeanRegistrationPhaseBuildItem beanRegistrationPhase,
             BuildProducer<NativeImageProxyDefinitionBuildItem> proxyDefinition) {
         checkClasspath(APACHE_HTTP_SERVICE, "apache-client");
```
Comment (ppalaga) [inline extensions/aws2-s3/deployment/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/Aws2S3Processor.java:90]: Isn't this class path check also a good candidate to move to the new support extension?

File: extensions/aws2-s3/deployment/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/Aws2S3Processor.java [hunk 2]
```diff
@@ -89,12 +89,6 @@ void client(BeanRegistrationPhaseBuildItem beanRegistrationPhase,
...
 
-        //Register Apache client as sync client
-        proxyDefinition.produce(
-                new NativeImageProxyDefinitionBuildItem("org.apache.http.conn.HttpClientConnectionManager",
-                        "org.apache.http.pool.ConnPoolControl",
-                        "software.amazon.awssdk.http.apache.internal.conn.Wrapped"));
-
         serviceProvider.produce(new ServiceProviderBuildItem(SdkHttpService.class.getName(), APACHE_HTTP_SERVICE));
```
Comment (ppalaga) [inline extensions/aws2-s3/deployment/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/Aws2S3Processor.java:92]: Similar as above: Isn't this serviceProvider also a good candidate to move to the new support extension?

Comment (oscerd) [reply extensions/aws2-s3/deployment/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/Aws2S3Processor.java:92]: Actually moving it there doesn't work, at least for me. I'll have to dig a bit more in this. This is the beginning.

File: extensions/aws2-s3/deployment/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/Aws2S3Processor.java (hunk 1 above)
Comment (oscerd) [reply extensions/aws2-s3/deployment/src/main/java/org/apache/camel/quarkus/component/aws2/s3/deployment/Aws2S3Processor.java:90]: Yes, but actually it doesn't seem to work when moved to commons. So lets have a skeleton for the moment and adding more extensions and then revisit the support extension, if it's ok for all, obviously

**Comment by oscerd** [comment]
The Fhir failing tests is unrelated

**Pull request merged**